plugins/discourse-github/app/controllers/discourse_github/pull_requests_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module DiscourseGithub
+  class PullRequestsController < ::ApplicationController
+    requires_plugin "discourse-github"
+
+    def status
+      owner = params[:owner]
+      repo = params[:repo]
+      number = params[:number]
+
+      if owner.blank? || repo.blank? || number.blank?
+        return(
+          render json: {
+                   error: I18n.t("discourse_github.errors.missing_parameters"),
+                 },
+                 status: :bad_request
+        )
+      end
+
+      if state = GithubPrStatus.fetch(owner, repo, number)
+        render json: { state: state }
+      else
+        render json: {
+                 error: I18n.t("discourse_github.errors.failed_to_fetch_pr_status"),
+               },
+               status: :bad_gateway
```

Comment:
I think using 502 "Bad Gateway" could be very confusing. Normally, 502s are a signal from intermediate proxies in front of Discourse, like NGINX & HAProxy.

This controller is not a transparent proxy to GitHub's API. It's adding auth, and creating its own response. I think it'll need caching as well, before we could consider enabling on any production site.

So, a 500 'server error' might be more appropriate here?